recipes/ffmpeg/all/conanfile.py

```diff
@@ -543,57 +546,103 @@ def opt_append_disable_if_set(args, what, v):
             ])
         if not self.options.with_programs:
             args.append("--disable-programs")
-        # since ffmpeg's build system ignores toolchain variables
-        if tools.get_env("AR"):
-            args.append("--ar={}".format(tools.get_env("AR")))
-        if tools.get_env("AS") and self.options.with_asm:
-            args.append("--as={}".format(tools.get_env("AS")))
-        if tools.get_env("CC"):
-            args.append("--cc={}".format(tools.get_env("CC")))
-        if tools.get_env("CXX"):
-            args.append("--cxx={}".format(tools.get_env("CXX")))
-        if tools.get_env("NM"):
-            args.append("--nm={}".format(tools.get_env("NM")))
-        if tools.get_env("RANLIB"):
-            args.append("--ranlib={}".format(tools.get_env("RANLIB")))
-        if tools.get_env("STRIP"):
-            args.append("--strip={}".format(tools.get_env("STRIP")))
-        extra_cflags = []
-        extra_ldflags = []
-        if is_apple_os(self) and self.settings.os.version:
-            extra_cflags.append(tools.apple_deployment_target_flag(
-                self.settings.os, self.settings.os.version))
-            extra_ldflags.append(tools.apple_deployment_target_flag(
-                self.settings.os, self.settings.os.version))
+        # since ffmpeg"s build system ignores CC and CXX
+        compilers_from_conf = self.conf.get("tools.build:compiler_executables", default={}, check_type=dict)
+        buildenv_vars = VirtualBuildEnv(self).vars()
+        nm = buildenv_vars.get("NM")
+        if nm:
+            args.append(f"--nm={unix_path(self, nm)}")
+        ar = buildenv_vars.get("AR")
+        if ar:
+            args.append(f"--ar={unix_path(self, ar)}")
+        if self.options.with_asm:
+            asm = compilers_from_conf.get("asm", buildenv_vars.get("AS"))
+            if asm:
+                args.append(f"--as={unix_path(self, asm)}")
+        strip = buildenv_vars.get("STRIP")
+        if strip:
+            args.append(f"--strip={unix_path(self, strip)}")
+        cc = compilers_from_conf.get("c", buildenv_vars.get("CC", self._default_compilers.get("cc")))
+        if cc:
+            args.append(f"--cc={unix_path(self, cc)}")
+        cxx = compilers_from_conf.get("cpp", buildenv_vars.get("CXX", self._default_compilers.get("cxx")))
+        if cxx:
+            args.append(f"--cxx={unix_path(self, cxx)}")
+        ld = buildenv_vars.get("LD")
+        if ld:
```

Comment:
There is an issue when cross building ffmpeg with the ld flag. In the old package, the ld was not set and I have the feeling, that this had some reasons. 

The problem is, when setting ld is that it that the build will fail with: `C compiler test failed.`

When checking the logs it seems, that the test build is trying to run gcc commands using the specified linker, which results in:
```
aarch64-linux-gnu-ld: unrecognized option '-Wl,-Bsymbolic,-znoexecstack'
```

 which definitely the wrong tool for that flags.
I'm not sure, if this is somehow explainable or a bug in the configuration stage of ffmpeg itself. Or maybe I'm doing something wrong. Maybe someone would need to reproduce the issue.
If the issue can be reproduced I would vote to remove the ld flag setting for the moment since it was not set in the old conanfile as well. At least for me, when removing the ld part, the cross compilation is working finally.

Reply:
```
x86_64-linux-gnu-ld: unrecognised emulation mode: 64
Supported emulations: elf_x86_64 elf32_x86_64 elf_i386 elf_iamcu elf_l1om elf_k1om
C compiler test failed.
```